scripts/server.py

```diff
@@ -550,66 +567,544 @@ def handle_client(conn, addr):
+                        # detection_response_status_json_decoded = json.loads(detection_response_status_json)
+
+                        # Check the response
+                        if detection_response_status_code == 200 or detection_response_status_code == 201 or detection_response_status_code == 202:
+                            # Extract the bird detection info to display in the output
+                            bird_detection_string = ""
+                            if 'detection' in detection_response_status_json:
+                                bird_detection_name = detection_response_status_json['detection']['species'][
+                                    'commonName']
+                                bird_detection_confidence = detection_response_status_json['detection']['confidence']
+                                bird_detection_timestamp = datetime.datetime.fromisoformat(
+                                    detection_response_status_json['detection']['timestamp'])
+                                bird_detection_time = bird_detection_timestamp.time()
+                                bird_detection_string = f"- {bird_detection_time}/{bird_detection_name}/{bird_detection_confidence}"
+
+                            if debug_birdweather_submissions:
+                                # Add in the JSON response if debugging just in case we might want to view it
+                                print(
+                                    f"BirdWeather Submission:: {retry_text}Detection Successfully Uploaded - status:{detection_response_status_code} json:{detection_response_status_json}",
+                                    flush=True)
+                            else:
+                                print(
+                                    f"BirdWeather Submission:: {retry_text}Detection Successfully Uploaded - status:{detection_response_status_code} {bird_detection_string}",
+                                    flush=True)
+                        else:
+                            print(
+                                f"BirdWeather Submission Error:: {retry_text}Something went wrong during detection submission - status: {detection_response_status_code} json:{detection_response_status_json}",
+                                flush=True)
+
+                        # Break the loop, if we reach here then were no exceptions to retry on
+                        # If the upload wasn't successful (non 2XX response, but also not 400+) for whatever reason, we won't retry
+                        # break
+
+                    except IndexError as idx_exec:
+                        print(
+                            f"BirdWeather Submission Error:: {retry_text}Detection POST - Unable to read response, something went wrong! - {idx_exec}",
+                            flush=True)
+                    except requests.exceptions.ConnectTimeout as conn_timeout_exec:
```

Comment:
here until 968: see above (covered by for, no need for individual treatment of these errors)

Reply:
some of these are gone now with unneeded response code removed (including IndexError). 918 is the similar to the original code with the extra debug line above it.